private/private.go

```diff
@@ -46,55 +49,183 @@ type PrivateTransactionManager interface {
 	DecryptPayload(payload common.DecryptRequest) ([]byte, *engine.ExtraMetadata, error)
 }
 
-func FromEnvironmentOrNil(name string) PrivateTransactionManager {
-	cfgPath := os.Getenv(name)
-	if cfgPath == "" {
-		return nil
-	}
-	if strings.EqualFold(cfgPath, "ignore") {
-		return &notinuse.PrivateTransactionManager{}
-	}
-	return MustNewPrivateTxManager(cfgPath)
+// This loads any config specified via the legacy environment variable
+func GetLegacyEnvironmentConfig() (engine.Config, error) {
+	return FromEnvironmentOrNil("PRIVATE_CONFIG")
 }
 
-func MustNewPrivateTxManager(cfgPath string) PrivateTransactionManager {
-	ptm, err := NewPrivateTxManager(cfgPath)
+func FromEnvironmentOrNil(name string) (engine.Config, error) {
+	cfgPath := os.Getenv(name)
+	cfg, err := engine.FetchConfigOrIgnore(cfgPath)
 	if err != nil {
-		panic(err)
+		return engine.Config{}, err
 	}
-	return ptm
+
+	return cfg, nil
 }
 
-func NewPrivateTxManager(cfgPath string) (PrivateTransactionManager, error) {
-	cfg, err := engine.LoadConfig(cfgPath)
-	if err != nil {
-		return nil, fmt.Errorf("unable to read %s due to %s", cfgPath, err)
+func InitialiseConnection(cfg engine.Config) error {
+	var err error
+	P, err = NewPrivateTxManager(cfg)
+	return err
+}
+
+func IsQuorumPrivacyEnabled() bool {
+	return isPrivacyEnabled
+}
+
+func NewPrivateTxManager(cfg engine.Config) (PrivateTransactionManager, error) {
+
+	if cfg.ConnectionType == engine.NoConnection {
+		log.Info("Running with private transaction manager disabled - quorum private transactions will not be supported")
+		return &notinuse.PrivateTransactionManager{}, nil
 	}
 
-	client := &engine.Client{
-		HttpClient: &http.Client{
-			Transport: unixTransport(cfg),
-		},
-		BaseURL: "http+unix://c",
+	client, err := createClient(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create connection to private tx manager due to: %s", err)
 	}
 
 	ptm, err := selectPrivateTxManager(client)
 	if err != nil {
-		return nil, fmt.Errorf("unable to connect to private tx manager using %s due to %s", cfgPath, err)
+		return nil, fmt.Errorf("unable to connect to private tx manager due to: %s", err)
 	}
+
+	isPrivacyEnabled = true
 	return ptm, nil
 }
 
+func createClient(cfg engine.Config) (*engine.Client, error) {
```

Comment:
It feels like next functions are not opinionated of ptm and thus could be moved under a common package and exported (e.g under `/common/http/client.go`, `/common/http/transport.go`, `/common/http/certificate.go`):
- createClient
- unixTransport
- httpTransport
- setHttpTransportToUseTLS
- loadRootCACerts
- loadRootCAFromFile

`struct Config` could also be moved to `/common/http/config.go`

Reply:
Done